fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/AdminUnitTest.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2016 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */package edu.unc.lib.dl.fcrepo4;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import com.hp.hpl.jena.rdf.model.Model;
+
+import edu.unc.lib.dl.fedora.ObjectTypeMismatchException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.PcdmModels;
+
+public class AdminUnitTest extends AbstractFedoraTest {
+	
+	private PID pid;
+	private AdminUnit unit;
+	
+	private PID collectionChildPid;
+	private PID workChildPid;
+
+	private CollectionObject collectionChildObj;
+	private WorkObject workChildObj;
+
+	@Before
+	public void init() {
+		initMocks(this);
+
+		pid = PIDs.get(UUID.randomUUID().toString());
+
+		unit = new AdminUnit(pid, repository, dataLoader);
+
+		collectionChildPid = PIDs.get(UUID.randomUUID().toString());
+		workChildPid = PIDs.get(UUID.randomUUID().toString());
+		when(repository.mintContentPid()).thenReturn(collectionChildPid).thenReturn(workChildPid);
+		
+		collectionChildObj = new CollectionObject(collectionChildPid, repository, dataLoader);
+		when(repository.createCollectionObject(any(PID.class), any(Model.class)))
+				.thenReturn(collectionChildObj);
+		
+		workChildObj = new WorkObject(workChildPid, repository, dataLoader);
```

Comment:
You could probably get away with mocking a WorkObject here instead of instantiating one, which is preferable for a unit test.  Also, I don't think repository.createWorkObject work would get called in this test?  Same suggestions apply to the CollectionObject being created at line 64.